### PR TITLE
DS-2949 Bump lodash package version

### PR DESCRIPTION
**Pull request by ildomar-grings**
## Overview

[DS-2949](https://loyaltynz.atlassian.net/browse/DS-2949)

## What is the change?
- Bump `lodash` version to fix critical vulnerabilities,  from `3.10.1` to `4.17.21`
- Bump `mocha` from `3.4.1` to `7.2.0`
- Bump `alchemy-ether` from `1.0.0` to `1.0.3`
- Introduce audit ignore list
- Update NPM Node Packages secure key: https://loyaltynz.atlassian.net/jira/servicedesk/projects/DS/queues/issue/DS-3514

## Notes

There is 1 critical vulnerability I could not fix without upgrading `Mocha` further:
```

┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Critical      │ flat vulnerable to Prototype Pollution                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ flat                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mocha [dev]                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mocha > yargs-unparser > flat                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/1089152                   │
└───────────────┴──────────────────────────────────────────────────────────────┘


┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ minimatch ReDoS vulnerability                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ minimatch                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ mocha [dev]                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ mocha > minimatch                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/1092637                   │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

[DS-2949]: https://loyaltynz.atlassian.net/browse/DS-2949?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,10 +22,17 @@ cache:
 before_install:
   - export CXX="g++-4.8"
 
+before_script:
+  - npm install -g npm@6.2.0
+  - npm install -g audit-filter
+
+script:
```
Comment (davidoram) [inline .travis.yml:29]: Can you verify that travis is still running the unit tests - I couldn't see any output relating to that on the travis builds?

**Review by davidoram — COMMENTED** [review]
One question around test output in travis - otherwise 👍

File: .travis.yml (hunk 1 above)
Comment (ildomar-grings) [reply .travis.yml:29]: Good question, that is intriguing. I will check what happened

Comment (ildomar-grings) [reply .travis.yml:29]: Tests trigger restored.
Good catch

https://app.travis-ci.com/github/LoyaltyNZ/alchemy-resource/builds/265323048

**Review by davidoram — APPROVED** [review]
GTG 🥇

**Pull request merged**